packages/paste-core/components/input-box/src/InputChevronWrapper.tsx

```diff
@@ -10,23 +56,31 @@ export interface InputChevronWrapperProps {
   element: BoxProps['element'];
 }
 
-export const InputChevronWrapper = React.forwardRef<HTMLDivElement, InputChevronWrapperProps>(
-  ({children, element = 'CHEVRON_WRAPPER'}, ref) => (
-    <Box
-      alignItems="center"
-      display="inline-flex"
-      position="absolute"
-      pointerEvents="none"
-      right="space30"
-      top="50%"
-      transform="translateY(-50%)"
-      zIndex="zIndex10"
-      element={element}
-      ref={ref}
-    >
-      {children}
-    </Box>
-  )
+const InputChevronWrapper = React.forwardRef<HTMLDivElement, InputChevronWrapperProps>(
+  ({children, element = 'CHEVRON_WRAPPER'}, ref) => {
+    const {disabled, readOnly, variant} = useInputBoxContext();
+    const state = disabled ? 'disabled' : readOnly ? 'readOnly' : 'default';
```

Comment:
Question: this logic in the component looks to be the same as the logic in the utility. Is there a chance they get out of sync?